### PR TITLE
Philips api

### PR DESCRIPTION
**Pull request by NikitaMikhaliuk**

### CODE REVIEW

File: src/core/interfaces/manager.interface.ts [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+  IPhAPIDataAccessServiceConfig,
+  IPhAPILightServiceConfig
+} from './app.interface';
 
 export namespace Manager {
+  export namespace Service {
+    export interface Config {
+        sequence: TID.Interfaces.Manager.Sequence.Config;
+    }
+    export interface GetServiceStoreConfig {
```
Comment (AndreyShuleiko) [inline src/core/interfaces/manager.interface.ts]: You can rename the this interface to the `APIConfig`.

File: src/core/interfaces/app.interface.ts [hunk 2]
```diff
@@ -4,3 +4,7 @@ export interface IPhAPILightServiceConfig {
     buildingId: string;
     auth: IPhAuth;
 }
+
+export interface IPhAPIDataAccessServiceConfig {
```
Comment (AndreyShuleiko) [inline src/core/interfaces/app.interface.ts]: You can create the namespace `Service` - `Light` : `Data` with `Config` interface.

File: src/core/services/data-access/data-access.service.base.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { ApiError } from '../../errors';
+
+/* Interfaces */
+import {
+    IPhAPIDataAccessServiceConfig,
+    IPhAPIMetricsReport,
+    IPhAPIMetricsReportOptions
+} from '../../interfaces';
+
+export class APIDataAccessServiceBase extends APIServiceBase {
```
Comment (AndreyShuleiko) [inline src/core/services/data-access/data-access.service.base.ts]: You can use `typescript` import scope for class name.

File: src/core/services/data-access/data-access.service.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+} from '../../interfaces';
+
+
+export class APIDataService extends APIDataAccessServiceBase {
+    private config: IPhAPIDataAccessServiceConfig;
+
+    private restClient: TIDClient.Clients.REST;
+    private soapClient: TIDClient.Clients.SOAP;
+
+    constructor (clientStore: any) {
```
Comment (AndreyShuleiko) [inline src/core/services/data-access/data-access.service.ts]: You can add the `private` access modifier.

File: src/core/services/data-access/data-access.service.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        super();
+        this.restClient = clientStore.rest;
```
Comment (AndreyShuleiko) [inline src/core/services/data-access/data-access.service.ts]: You can use `clientStore.rest` directly.

File: src/core/services/data-access/data-access.service.ts [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+          metricType: opts.metricType
+          },
+        },
+      };
+      switch (opts.metricType) {
+        case 'Occupancy': {
+          return this.getOccupancyReport(reqOpts);
+        }
+        default:
+
```
Comment (AndreyShuleiko) [inline src/core/services/data-access/data-access.service.ts]: You can add the `TODO: ` comments.

File: src/core/interfaces/manager.interface.ts (hunk 1 above)
Comment (NikitaMikhaliuk) [reply src/core/interfaces/manager.interface.ts]: changed

File: src/core/interfaces/app.interface.ts (hunk 2 above)
Comment (NikitaMikhaliuk) [reply src/core/interfaces/app.interface.ts]: changed

File: src/core/interfaces/philips.interface.ts [hunk 7]
```diff
@@ -61,3 +61,33 @@ export interface IPhAPIAreaLuminaireLevels {
     areaId: number;
     luminaireLevels: IPhLuminaireLevel[];
 }
+
+//getMetricsReport
+export interface IPhAPIMetricsReportOptions {
```
[the remark this answers is not in the document]
Comment (NikitaMikhaliuk) [reply src/core/interfaces/philips.interface.ts]: changed

File: src/core/managers/service.manager.ts [hunk 8]
```diff
@@ -1,48 +1,47 @@
...
+    */
+    public getServiceStore (config: Interfaces.Manager.Service.GetServiceStoreConfig): Store {
+      const serviceStore = new Store();
+
+      const clientStore = this.clientManager.getClientStore();
+      const apiDataAccessService = new APIDataService(clientStore);
+      apiDataAccessService.initService(config.dataService)
+      const apiLightService = new APILightService(clientStore);
+      apiLightService.initService(config.lightService);
+      serviceStore.apiDataAccessService = apiDataAccessService;
```
[the remark this answers is not in the document]
Comment (NikitaMikhaliuk) [reply src/core/managers/service.manager.ts]: changed

File: src/core/services/data-access/data-access.service.base.ts (hunk 3 above)
Comment (NikitaMikhaliuk) [reply src/core/services/data-access/data-access.service.base.ts]: changed

**Pull request merged**